includes/admin/class-wp-job-manager-promoted-jobs.php

```diff
@@ -0,0 +1,181 @@
+		add_action( 'manage_job_listing_posts_custom_column', [ $this, 'promoted_jobs_custom_columns' ], 2 );
+		add_action( 'admin_footer', [ $this, 'promoted_jobs_admin_footer' ] );
+	}
+
+	/**
+	 * Add a column to the job listings admin page.
+	 *
+	 * @param array $columns Columns.
+	 * @return array
+	 */
+	public function promoted_jobs_columns( $columns ) {
+		$columns['promoted_jobs'] = __( 'Promote', 'wp-job-manager' );
+		return $columns;
+	}
+
+	/**
+	 * Check if a job is promoted.
+	 *
+	 * @param int $post_id
+	 * @return boolean
+	 */
+	public function is_promoted( $post_id ) {
+		$promoted = get_post_meta( $post_id, '_promoted', true );
+		return $promoted ? true : false;
+	}
+
+	/**
+	 * Handle display of new column
+	 *
+	 * @param  string $column
+	 */
+	public function promoted_jobs_custom_columns( $column ) {
+		global $post;
+		if ( 'promoted_jobs' === $column ) {
+			if ( $this->is_promoted( $post->ID ) ) {
+				echo '
+					Live
+					<br />
+					<a href="#">Edit promotion</a>
+					<br />
+					<a href="#">Deactivate</a>
+				';
+			} else {
+				echo '<button class="promote_job button button-primary" data-post=' . esc_attr( $post->ID ) . '>Promote</button>';
+			}
+		}
+	}
+	/**
+	 * Store the promoted jobs template from wpjobmanager.com
+	 *
+	 * TODO: we need to fetch this from wpjobmanager.com and store in cache for X amount of time
+	 * We should also have a fallback in case the API call fails.
+	 * We need to have a fallback here because we can't use a `dialog` element inside the template
+	 *
+	 * @return string
+	 */
+	public function get_promote_jobs_template() {
+		return '
+		<dialog class="promoteJobsDialog" id="promoteDialog">
+			<form method="dialog">
+				<button type="submit" autofocus>X</button>
+			</form>
+			<promote-job-modal>
+				<div slot="column-left" class="promote-job-modal-column-left">
+					<h2 slot="promote-heading">
+						Promote your job on our partner network.
+					</h2>
+
+					<div slot="price" class="promote-job-modal-price">
+						<div class="price-text">Starting From</div>
+						<span>$83.00</span>
+					</div>
+
+					<div slot="promote-list">
+						<ul>
+							<li>Your ad will get shared on our Partner Network</li>
+							<li>Featured on jobs.blog for 7 days</li>
+							<li>Featured on our weekly email blast</li>
+						</ul>
+					</div>
+
+					<div slot="buttons" class="promote-buttons-group">
+						<button class="button button-primary" type="submit">Promote your job</button>
+						<button class="button button-secondary" type="submit">Learn More</button>
+					</div>
+				</div>
+				<div slot="column-right" class="promote-job-modal-column-right">
+					<img src="https://d.pr/i/4PgTqN+">
+				</div>
+			</promote-job-modal>
+		</dialog>';
+	}
+
+	/**
+	 * Output the promote jobs template
+	 *
+	 * @return void
+	 */
+	public function promoted_jobs_admin_footer() {
+		echo '
+			<template id="promote-job-template">
+				<slot name="column-left" class="column-left">
```

Comment:
We need to also update the classes here for the fallback.

Reply:
The template doesn't include the dialog - so I don't think we do in this case?  I found out that you can't have dialogs in templates, they just don't seem to work.

Reply:
Ah, I meant about classes like the `class="column-left"` that we updated.

Reply:
Gotcha!  Done in a44c9bcd56323db227551d8426c1467637876878